download-basket/src/main/java/downloadbasket/helpers/DownloadServices.java

```diff
@@ -0,0 +1,277 @@
+			}
+			
+			if(ldz.getGetFeatureInfoRequest().isEmpty()){
+				return null;
+			}
+			
+			LOGGER.debug("WFS URL: " + ldz.getWFSUrl());
+			LOGGER.debug("-- filtter: " + ldz.getGetFeatureInfoRequest());
+			final URL url = new URL(ldz.getWFSUrl() + ldz.getGetFeatureInfoRequest());
+			
+			conn = (HttpURLConnection) url.openConnection();
+			conn.setRequestMethod("GET");
+			conn.setConnectTimeout(600000);
+
+			conn.connect();
+		
+			String filename = UUID.randomUUID().toString();
+			istream = conn.getInputStream();
+			
+			String strTempDir = ldz.getTemporaryDirectory();
+			
+			File dir0 = new File(strTempDir);
+			dir0.mkdirs();
+			
+			String slashType = "";
+			String myFullFileName = dir0.getName();
+			if (myFullFileName.lastIndexOf("\\")> 0) {
+				slashType = "\\";
+			} else {
+				slashType = "/";
+			}
+			
+			realFileName = strTempDir + slashType + filename + ".zip";
+			
+			returnFileName = strTempDir + slashType + filename + ".zip";
+			ostream = new FileOutputStream(realFileName);
+			
+
+			final byte[] buffer = new byte[8*1024];
+		
+			while (true)
+		    {
+				int len = istream.read(buffer);
+				if (len <= 0)
+		        { 
+		    		break; 
+		    	}
+				ostream.write(buffer, 0, len);
+		    }			
+			
+			if(!isValid(new File(realFileName))){
+				ErrorReportDetails erd = new ErrorReportDetails();
+				erd.setErrorFileLocation(realFileName);
+				erd.setWfsUrl(ldz.getWFSUrl());
+				erd.setXmlRequest(ldz.getGetFeatureInfoRequest());
+				erd.setUserEmail(ldz.getUserEmail());
+				erd.setLanguage(ldz.getLanguage());
+				sendErrorReportToEmail(erd);
+				sendErrorReportToUserEmail(erd);
+			}
+			
+		} catch(Exception ex){
+			ex.printStackTrace();
+		} finally{
+			if(writer!=null){
+				writer.close();
+			}
+			if(conn!=null){
+				conn.disconnect();
+			}
+			if(istream!=null){
+				istream.close();
+			}
+			if(ostream!=null){
+				ostream.close();
+			}
+		}
+		return returnFileName;
+	}
+	
+	/**
+	 * Check at is zipfile valid.
+	 * @param file zip file
+	 * @return
+	 */
+	static boolean isValid(final File file) {
+	    ZipFile zipfile = null;
+	    
+	    try {
+	        zipfile = new ZipFile(file);
+	        return true;
+	    } catch (ZipException e) {
+	        return false;
+	    } catch (IOException e) {
+	        return false;
+	    } finally {
+	        try {
+	            if (zipfile != null) {
+	                zipfile.close();
+	                zipfile = null;
+	            }
+	        } catch (IOException e) {
+	        }
+	    }
+	}
+	
+	/***
+	 * Send error report to support email.
+	 * @param xmlRequest xml request
+	 * @param wfsUrl wfs url
+	 */
+	
+	/**
+	 * Send error report to support email.
+	 * @param errorDetails error report details
+	 */
+	private void sendErrorReportToEmail(ErrorReportDetails errorDetails) {
+		try {
+			String msg = "";			
+			msg += "<b>Tapahtui virhe WFS-latauksessa</b><br/><br/>WFS-url:<br/>" + errorDetails.getWfsUrl() + "<br/><br/>"+
```

Comment:
See https://github.com/oskariorg/oskari-server/blob/master/control-users/src/main/java/fi/nls/oskari/control/users/service/MailSenderService.java 

Preferrably create a new service (maybe even in service-base) that both functionalities can use to send email messages. At least use the same properties for smtp host and such.